### PR TITLE
Feature/solflare integration

### PR DESCRIPTION
**Pull request by tsmbl**
**Context:** 

1. There're old monitoring services that may send notifications to on-chain threads directly via dialect protocol
2. There're new monitoring services that may send notifications to on-chain threads via dialect sdk
3. There're new monitoring services that may send notifications to off-chain threads via dialect sdk, in this case notifications are sent to data service and dispatched there

**Solution:**
1. On chain messages are caught by https://github.com/dialectlabs/message-webhook-service and forwarded to solflare, this covers cases **(1)** and **(2)** from context section
2. Off-chain notifications are forwarded to solflare from data service if user has subscribed to wallet notifications from dapp and has off-chain thread, covers case **(3)** from context section

Also, @kevindingens did an experiment with EventEmitter and Events, thought this will be needed to cover solflare use-case.  I didn't use it in a final solution, but decided to leave the changes, lmk what do you think about it

### CODE REVIEW

File: src/dapp-message/dapp-message.service.ts [hunk 1]
```diff
@@ -192,4 +190,50 @@ export class DappMessageService {
...
+    message: string,
+    receivers: PublicKey[],
+    sender: DappPrincipal,
+  ) {
+    const dialects = await this.findDialectsToSendMessage(receivers, sender);
+    const receiverPublicKeys = dialects
+      .flatMap((it) => it.members)
+      .map((it) => it.wallet.publicKey)
+      .filter((it) => it !== sender.wallet.publicKey);
+    await this.solflare.send(message, sender.dapp, receiverPublicKeys);
```
Comment (zhelezkov) [inline src/dapp-message/dapp-message.service.ts:204]: maybe we shouldn't block the execution?

Comment (zhelezkov) [reply src/dapp-message/dapp-message.service.ts:204]: I'm just thinking that solflare probably not only one use case for webhooks, maybe it makes sense to abstract from this somehow, and dispatch across all injected services?
or maybe it's out of scope

File: src/dapp-message/dapp-message.service.ts [hunk 2]
```diff
@@ -47,7 +52,8 @@ export class DappMessageService {
     private readonly telegram: TelegramService,
     private readonly mail: MailService,
     private readonly sms: SmsService,
-    private readonly dialect: DialectService,
+    private readonly solflare: SolflareService,
```
Comment (zhelezkov) [inline src/dapp-message/dapp-message.service.ts:55]: shouldn't it be optional depending on env vars configuration?

**Review by zhelezkov — APPROVED** [review]
overall looks good. some minor question, but don't insist on fixing rn

File: src/dapp-message/dapp-message.service.ts (hunk 2 above)
Comment (tsmbl) [reply src/dapp-message/dapp-message.service.ts:55]: yep, this is actually done in this way but hidden in factory 
image

image

**Pull request merged**